Improve documentation

cwlacewe opened the pull request:
Closes #285 

This PR pulls in our Wiki documentation and generates a page using mkdocs.
For each merge to develop, the documentation will be updated.

cwlacewe committed:
Transfer wiki to mkdocs

Signed-off-by: cwlacewe <chaunte.w.lacewell@intel.com>

cwlacewe committed:
Add doc deploy to workflow

Signed-off-by: cwlacewe <chaunte.w.lacewell@intel.com>

cwlacewe committed:
Change trigger of update docs to push

Signed-off-by: cwlacewe <chaunte.w.lacewell@intel.com>

cwlacewe committed:
Remove doc artifact

Signed-off-by: cwlacewe <chaunte.w.lacewell@intel.com>

cwlacewe committed:
TESTING ONLY: Add push to branch as trigger

Signed-off-by: cwlacewe <chaunte.w.lacewell@intel.com>

cwlacewe committed:
mod update doc workflow, add dark/light mode to mkdocs.yml

Signed-off-by: cwlacewe <chaunte.w.lacewell@intel.com>

github-actions[bot] committed:
Automated coverage update

cwlacewe committed:
Change theme to readthedocs and add DeepWiki links

Signed-off-by: cwlacewe <chaunte.w.lacewell@intel.com>

cwlacewe committed:
Merge branch 'develop' into cwl-docs

Signed-off-by: cwlacewe <chaunte.w.lacewell@intel.com>

github-actions[bot] committed:
Automated coverage update

cwlacewe committed:
Organize mkdocs and add req file

Signed-off-by: cwlacewe <chaunte.w.lacewell@intel.com>

cwlacewe committed:
Merge branch 'cwl-docs' of https://github.com/IntelLabs/vdms into cwl-docs

cwlacewe committed:
Fix workflow

Signed-off-by: cwlacewe <chaunte.w.lacewell@intel.com>

github-actions[bot] committed:
Automated coverage update

cwlacewe committed:
Update guides

Signed-off-by: cwlacewe <chaunte.w.lacewell@intel.com>

cwlacewe committed:
Rename guide section

Signed-off-by: cwlacewe <chaunte.w.lacewell@intel.com>

cwlacewe committed:
Format updates 1 and re-structure files

Signed-off-by: cwlacewe <chaunte.w.lacewell@intel.com>

cwlacewe committed:
Format update

Signed-off-by: cwlacewe <chaunte.w.lacewell@intel.com>

cwlacewe committed:
Pull latest wiki updates

Signed-off-by: cwlacewe <chaunte.w.lacewell@intel.com>

cwlacewe committed:
Change references to wiki to site https://intellabs.github.io/vdms/; remove INSTALL.md and use version in docs/

Signed-off-by: cwlacewe <chaunte.w.lacewell@intel.com>

cwlacewe committed:
Cleanup; Update trigger for pushes to develop and master

Signed-off-by: cwlacewe <chaunte.w.lacewell@intel.com>

github-actions[bot] committed:
Automated coverage update

cwlacewe committed:
Merge branch 'develop' into cwl-docs

github-actions[bot] committed:
Automated coverage update

cwlacewe committed:
Spelling fixes and add conditional statement to job pushing documentation (added test trigger for branch)

Signed-off-by: Lacewell, Chaunte W <chaunte.w.lacewell@intel.com>

cwlacewe committed:
fix workflow condition

Signed-off-by: Lacewell, Chaunte W <chaunte.w.lacewell@intel.com>

cwlacewe committed:
fix workflow command

Signed-off-by: Lacewell, Chaunte W <chaunte.w.lacewell@intel.com>

cwlacewe committed:
fix workflow command

Signed-off-by: Lacewell, Chaunte W <chaunte.w.lacewell@intel.com>

cwlacewe committed:
fix workflow command

Signed-off-by: Lacewell, Chaunte W <chaunte.w.lacewell@intel.com>

cwlacewe committed:
fix workflow command

Signed-off-by: Lacewell, Chaunte W <chaunte.w.lacewell@intel.com>

cwlacewe committed:
fix workflow command

Signed-off-by: Lacewell, Chaunte W <chaunte.w.lacewell@intel.com>

cwlacewe committed:
remove test text

Signed-off-by: Lacewell, Chaunte W <chaunte.w.lacewell@intel.com>

cwlacewe committed:
Remove test trigger

Signed-off-by: Lacewell, Chaunte W <chaunte.w.lacewell@intel.com>

keirafadams commented on docs/blocks/Block-collections.md:
Did we ever actually integrate collections? If not, we can probably just quietly nuke this.

cwlacewe replied:
All collection tests are skipped so omitted from docs site for now

rv355 reviewed (dismissed):
No Next or Back links for the links in the "Commands" sub-pages. For example: https://intellabs.github.io/vdms/commands/basic_building_blocks_index/.

keirafadams reviewed (dismissed):
Approved, nothing leaping out. Only minor one is if we dont officially support collections is we should probably quietly drop it.

cwlacewe committed:
Remove collections reference

Signed-off-by: Lacewell, Chaunte W <chaunte.w.lacewell@intel.com>

cwlacewe committed:
Removed *_index.md pages under commands to allow easier navigation with next and prev buttons: Fix incorrect link to Kubernetes guide

Signed-off-by: Lacewell, Chaunte W <chaunte.w.lacewell@intel.com>

cwlacewe commented:
> No Next or Back links for the links in the "Commands" sub-pages. For example: https://intellabs.github.io/vdms/commands/basic_building_blocks_index/.

Removed index pages and converted commands to subpages. They are now individually available in sidebar and the Next and Back buttons are enabled

github-actions[bot] committed:
Automated coverage update

keirafadams reviewed (approved):
Looks good
